src/main/java/seedu/address/logic/parser/exceptions/TooManyPrefixesException.java

```diff
@@ -1,9 +1,19 @@
 package seedu.address.logic.parser.exceptions;
 
-import seedu.address.logic.parser.Prefix;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PREAMBLE;
 
-public class TooManyPrefixesException extends IllegalPrefixException {
-    public TooManyPrefixesException(Prefix prefix) {
-        super(prefix, String.format("Too many arguments: %s!\n", prefix) + "%s");
+import seedu.address.logic.parser.CommandArgument;
+import seedu.address.logic.parser.CommandSpecification;
+
+public class TooManyPrefixesException extends IllegalCommandArgumentException {
+
+    /**
+     * Creates an exception representing the case where too many prefixes are supplied as arguments to a command.
+     * @param argument The argument with too many prefixes
+     * @param specs The specifications of the command
+     */
+    public TooManyPrefixesException(CommandArgument argument, CommandSpecification specs) {
+        super(String.format("Too many arguments for the prefix %s.", argument.getPrefix()), specs);
+        assert !argument.getPrefix().equals(PREFIX_PREAMBLE);
```

Comment:
Given that PREFIX_PREAMBLE is a required single, this assert is probably useless. 

But that's a good thing i guess LOL. It's just interesting to put this here imo

Reply:
#defensiveprogramming